samtranslator/model/eventsources/push.py

```diff
@@ -171,18 +208,43 @@ def to_cloudformation(self, **kwargs):
         events_rule = EventsRule(self.logical_id)
         events_rule.EventBusName = self.EventBusName
         events_rule.EventPattern = self.Pattern
-        events_rule.Targets = [self._construct_target(function)]
+        source_arn = events_rule.get_runtime_attr("arn")
+
+        dlq_queue_arn = None
```

Comment:
This block of code is copied into 4 different places. If they are all the same, they should be combined somehow (object, method, etc). The way it is currently written will make it very difficult to maintain parity when one thing gets changed in one place.

Reply:
Sounds good, addressed in the later revision